src/data/datasets/representation/unsupervised_contrastive_dataset.py

```diff
@@ -0,0 +1,57 @@
+from typing import Union, Optional
+
+import torch
+from albumentations import BasicTransform
+from albumentations.core.composition import BaseCompose
+
+from src.registry import DATASETS
+from src.data.datasets.base import ImageDataset
+
+
+@DATASETS.register_class
+class UnsupervisedContrastiveDataset(ImageDataset):
+    """A generic dataset for image contrastive task"""
```

Comment:
A dataset for unsupervised contrastive task (one image is transformed twice so that they are positive to each other)